BSP/STM32F413/Src/BSP_Contactor.c

```diff
@@ -45,20 +123,32 @@ void BSP_Contactor_On(void) {
  * @param   None
  * @return  None
  */
-void BSP_Contactor_Off(void) {
-    GPIO_WriteBit(GPIOB, GPIO_Pin_0, Bit_RESET);	// Set pin 0 low
+void BSP_Contactor_Off(CONT_CHOICE contactorChoice) {
+    // set output pins LOW
+	if (contactorChoice & ARRAY_CONTACTOR) {
+		GPIO_WriteBit(C1_PORT, GPIO_Pin_0, Bit_RESET);
+	}
+	if (contactorChoice & HVHIGH_CONTACTOR) {
+		GPIO_WriteBit(C2_PORT, GPIO_Pin_4, Bit_RESET);
+	}
+	if (contactorChoice & HVLOW_CONTACTOR) {
+		GPIO_WriteBit(C3_PORT, GPIO_Pin_0, Bit_RESET);
+	}
 }
 
 /**
  * @brief   Gets the state of the Contactor switch from one of its AUX pins.
  * @param   None
  * @return  0 if contactor is off/open, 1 if on/closed
  */
-bool BSP_Contactor_GetState(void) {
-    // The AUX pin we have connected to is negative logic
-    if((GPIOB->IDR & GPIO_Pin_1) >> 1) {
-		return 0;
-	} else {
-		return 1;
+bool BSP_Contactor_GetState(CONT_CHOICE contactorChoice) {
```

Comment:
should probably specify that you can only get the state of one contactor; if you pass ```ALL_CONTACTORS``` it'll just return the state for ```ARRAY_CONTACTOR```